test/wallet-standalone-test.js

```diff
@@ -0,0 +1,86 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+'use strict';
+
+const assert = require('./util/assert');
+const FullNode = require('../lib/node/fullnode');
+const SPVNode = require('../lib/node/spvnode');
+const WalletNode = require('../lib/wallet/node');
+
+const fullNode = new FullNode({
+  network: 'regtest',
+  memory: true
+});
+
+const prunedNode = new FullNode({
+  network: 'regtest',
+  memory: true,
+  prune: true
+});
+
+const spvNode = new SPVNode({
+  network: 'regtest',
+  memory: true
+});
+
+const wallet = new WalletNode({
+  network: 'regtest',
+  memory: true
+});
+
+describe('Standalone Wallet', function() {
+  describe('With full node', function() {
+    before(async () => {
+      await fullNode.open();
+      await wallet.open();
+      await new Promise(r => setTimeout(r, 100));
```

Comment:
I tried waiting for a `'connect'` event here, but it fires inconsistently (sometimes BEFORE `wallet.open()` resolves). Went down the rabbit hole all the way through `vendor/faye-websocket` and couldn't figure out a better strategy.

Reply:
You can setup listener before `open`, but wait after `.open`.

If connect happened before, it will be resolved. If not it will resolve after.

Reply:
hows this? https://github.com/bcoin-org/bcoin/pull/749/commits/d87276f5db83c5f7c6438fcba370b59c29866eab